sdk/communication/Azure.Communication.Administration/samples/Sample2_PhoneNumberAdministrationClient.md

```diff
@@ -0,0 +1,215 @@
+            PrintLocationOption(subLocationOption);
+    }
+}
+PrintLocationOption(locationOptionsResponse.Value.LocationOptions);
+```
+
+## Get area codes
+
+Fetching area codes for geographic phone plans will require the the location options queries set. You must include the chain of geographic locations traversing down the location options object returned by the GetPhonePlanLocationOptions.
+
+```C# Snippet:GeographicalAreaCodes
+var locationOptionsResponse = client.GetPhonePlanLocationOptions(countryCode, geographicPhonePlanGroupId, geographicPhonePlanId);
+var state = locationOptionsResponse.Value.LocationOptions.Options.First();
+
+var locationOptionsQueries = new List<LocationOptionsQuery>
+{
+    new LocationOptionsQuery
+    {
+        LabelId = "state",
+        OptionsValue = state.Value
+    },
+    new LocationOptionsQuery
+    {
+        LabelId = "city",
+        OptionsValue = state.LocationOptions.First().Options.First().Value
+    }
+};
+
+var areaCodes = client.GetAllAreaCodes(geographicPhonePlan.LocationType.ToString(), countryCode, geographicPhonePlan.PhonePlanId, locationOptionsQueries);
+
+foreach (var areaCode in areaCodes.Value.PrimaryAreaCodes)
+{
+    Console.WriteLine($"Primary area code: {areaCode}");
+}
+```
+
+Area codes for toll free phone plans can be found in the plan.
+
+```C# Snippet:TollFreePlanAreCodes
+var phonePlans = client.GetPhonePlans(countryCode, tollFreePhonePlanGroupId, locale);
+var tollFreePhonePlan = phonePlans.First();
+
+foreach (var areaCode in tollFreePhonePlan.AreaCodes)
+{
+    Console.WriteLine($"Area code: {areaCode}");
+}
+```
+
+## Reserve phone numbers
+
+Phone numbers need to be reserved for purchase. Reservation is a long running operation that can be started by CreateReservationOptions function that returns an PhoneNumberReservationOperation object. PhoneNumberReservationOperation can be used to update status of the operation and to check for completeness.
+
+```C# Snippet:ReservePhoneNumbers
+var reservationName = "My reservation";
+var reservationDescription = "reservation description";
+var reservationOptions = new CreateReservationOptions(reservationName, reservationDescription, new[] { phonePlanId }, areaCode);
+reservationOptions.Quantity = 1;
+
+var reservationOperation = client.StartReservation(reservationOptions);
+
+while (true)
+{
+    reservationOperation.UpdateStatus();
+    if (reservationOperation.HasCompleted)
+    {
+        break;
+    }
+
+    Thread.Sleep(1000);
+}
+```
+
+## Persist reserve phone numbers operation
+
+You can persist phone number reservation operation Id so that you can get back later to check operation status.
+
+```C# Snippet:PersistReservePhoneNumbersOperation
+var reservationId = reservationOperation.Id;
+
+// persist reservationOperationId and then continue with new operation
+
+var newPhoneNumberReservationOperation = new PhoneNumberReservationOperation(client, reservationId);
```

Comment:
Can we shorten the name? Maybe just `operation` or `reserveOperation`.

The context makes it clear what the operation is about and it can't be confused with another operation, that's why I think a short and less precise name reads better.

Reply:
+1 on `reserveOperation`

Reply:
+1 on `reserveOperation`

Reply:
I made it newReserveOperation to stress that is a new object created with reservationId
Also there is reserveOperation variable in the same test where code snippet is coming

Reply:
Usually one would only prefix variable names with `new` if they are a new replacement for something existing, for example an event handler with `newValue`. Otherwise every variable that holds a constructed object would start with `new`. In general, we're preferring short names over encoding types in the name, which is a practice that comes from Hungarian notation among others and is now discouraged.

To avoid a name clash with the other `reserveOperation` you can extract a function to have a new context.

Reply:
Renamed it to reserveOperation